net/interfaces/interfaces.go

```diff
@@ -117,17 +121,39 @@ func LocalAddresses() (regular, loopback []netaddr.IP, err error) {
 				if tsaddr.IsTailscaleIP(ip) {
 					continue
 				}
-				if ip.IsLinkLocalUnicast() {
-					continue
-				}
 				if ip.IsLoopback() || ifcIsLoopback {
 					loopback = append(loopback, ip)
+				} else if ip.Is4() && ip.IsLinkLocalUnicast() {
+					// In AWS Lambda, NAT for 169.254.x.x is the
+					// only connectivity.
+					linklocal4 = append(linklocal4, ip)
+
+					// We know of no cases where the IPv6 fe80:: addresses
+					// are used to provide WAN connectivity. It is also very
+					// common for users to have no IPv6 WAN connectivity,
+					// but their OS supports IPv6 so they have an fe80::
+					// address. We don't report IPv6 LL to Control.
+				} else if ip.Is6() && tsaddr.IsULA(ip) {
```

Comment:
but now you're including Tailscale's own IPv6 ULA addresses in linklocal6, and thus including them as a "regular" address (and thus a `HaveV6`)?

This would totally alter our working-IPv6 stats from the true number to nearly 100% if we start including everybody's self Tailscale IPv6 address?

Reply:
The conditional on line 121 will skip processing Tailscale addresses. IsTailscaleIP() checks TailscaleULARange().Contains(ip)

Reply:
ah, phew.